### PR TITLE
Add Stream.bracketFull

### PR DESCRIPTION
**Pull request by mpilquist**
This was relatively straightforward, but I don't know how to implement `translate`, which naively needs a `G ~> F` now. Thoughts?

### CODE REVIEW

**Comment by SystemFw** [comment]
> which naively needs a G ~> F now.

is this needed to translate `Poll`?

File: core/shared/src/main/scala/fs2/Pull.scala [hunk 1]
```diff
@@ -818,8 +819,9 @@ object Pull extends PullLowPriority {
       self match {
         // safe to cast, used in translate only
         // if interruption has to be supported concurrent for G has to be passed
-        case a: Acquire[F, r] =>
-          Acquire[G, r](fK(a.resource), (r, ec) => fK(a.release(r, ec)))
```
Comment (mpilquist) [inline core/shared/src/main/scala/fs2/Pull.scala:822]: We're given a `Poll[G]`, `Poll[F] => F[R]`, and `F ~> G` here.

**Comment by mpilquist** [comment]
> is this needed to translate Poll?

Yeah, that was my thought, though we'd need to add `imapK` to `Poll`.

**Comment by mpilquist** [comment]
On second thought, adding `Resource.makeMask` should introduce the same issue in CE3.

**Comment by SystemFw** [comment]
I'm wondering about the impact on real world scenarios, especially doobie

**Pull request closed without merging**